internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+	}
+	if internaltypes.IsDefined(model.RefreshTokenRollingInterval) && model.RefreshTokenRollingIntervalType.ValueString() != "OVERRIDE_SERVER_DEFAULT" {
+		resp.Diagnostics.AddError("refresh_token_rolling_interval_type must be configured to \"OVERRIDE_SERVER_DEFAULT\" to modify the refresh_token_rolling value.", "")
+	}
+
+	//  Client Auth Defined
+	var clientAuthDefined bool
+	var clientAuthAttributes map[string]attr.Value
+	if internaltypes.IsDefined(model.ClientAuth) {
+		clientAuthDefined = true
+		clientAuthAttributes = model.ClientAuth.Attributes()
+		if internaltypes.IsDefined(clientAuthAttributes["type"]) {
+			clientAuthType := clientAuthAttributes["type"].(types.String).ValueString()
+			switch clientAuthType {
+			case "PRIVATE_KEY_JWT":
+				if !internaltypes.IsNonEmptyObj(model.JwksSettings) {
+					resp.Diagnostics.AddError("jwks_settings must be defined when client_auth is configured to \"PRIVATE_KEY_JWT\".", "")
+				}
+			case "CERTIFICATE":
+				if !internaltypes.IsDefined(clientAuthAttributes["client_cert_subject_dn"]) || !internaltypes.IsDefined(clientAuthAttributes["client_cert_issuer_dn"]) {
+					resp.Diagnostics.AddError("client_cert_subject_dn and client_cert_issuer_dn must be defined when client_auth is configured to \"CERTIFICATE\".", "")
+				}
+			}
+		}
+	} else {
+		clientAuthDefined = false
+	}
+
+	// Grant Types Validation
+	// grant_types is required, don't need nil check here
+	var hasCibaGrantType bool
+	for _, grantType := range model.GrantTypes.Elements() {
+		grantTypeVal := grantType.(types.String).ValueString()
+		if grantTypeVal == "CLIENT_CREDENTIALS" {
+			if clientAuthDefined {
+				clientAuthType := clientAuthAttributes["type"].(types.String).ValueString()
+				clientAuthSecret := clientAuthAttributes["secret"].(types.String).ValueString()
+				if clientAuthType != "NONE" && clientAuthSecret == "" {
+					resp.Diagnostics.AddError("client_auth must be defined when \"CLIENT_CREDENTIALS\" is included in grant_types.", "")
+				}
+			} else if !clientAuthDefined {
+				resp.Diagnostics.AddError("client_auth must be defined when \"CLIENT_CREDENTIALS\" is included in grant_types.", "")
+			}
+		}
+		if grantTypeVal == "CIBA" {
+			hasCibaGrantType = true
+		}
+	}
+
+	// CIBA Validation
+	if !hasCibaGrantType && (internaltypes.IsDefined(model.CibaDeliveryMode) ||
+		internaltypes.IsDefined(model.CibaNotificationEndpoint) ||
+		internaltypes.IsDefined(model.CibaPollingInterval) ||
+		internaltypes.IsDefined(model.CibaRequireSignedRequests) ||
+		internaltypes.IsDefined(model.CibaRequestObjectSigningAlgorithm) ||
+		internaltypes.IsDefined(model.CibaUserCodeSupported)) {
+		resp.Diagnostics.AddError("ciba attributes can only be configured when \"CIBA\" is included in grant_types.", "")
+	}
+	if hasCibaGrantType &&
+		(internaltypes.IsDefined(model.CibaDeliveryMode) &&
```

Comment:
You shouldn't need this IsDefined check, since ValueString will just return "" if something isn't defined.